src/main/java/opendota/Parse.java

```diff
@@ -307,6 +314,8 @@ else if (cle.getType() == DOTA_COMBATLOG_TYPES.DOTA_COMBATLOG_GOLD) {
             else if (cle.getType() == DOTA_COMBATLOG_TYPES.DOTA_COMBATLOG_XP) {
                 combatLogEntry.xp_reason = cle.getXpReason();
             }
+            combatLogVisitors.visit(combatLogEntry);
```

Comment:
I think it would be safer to return output and assign properties to entry here.

Mutating state in the visit() function might be hard to track down.